### PR TITLE
DAL layer for user entity

### PR DESCRIPTION
**Pull request by KiSSel3**
- Created user entity
- Added interface and its implementation for working with users table
- Created validator for key user fields

### CODE REVIEW

**Comment by KiSSel3** [comment]
- Added a .idea folder to .gitignore

File: OnlineShop/OnlineShop.DAL/Validators/UserValidator.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+using FluentValidation;
+using OnlineShop.DAL.Entities.Implementations;
+
+namespace OnlineShop.DAL.Validators;
+
+public class UserValidator : AbstractValidator<User>
```
Comment (mebry) [inline OnlineShop/OnlineShop.DAL/Validators/UserValidator.cs]: Remove the validators from this layer. They should be on the BLL layer

File: OnlineShop/OnlineShop.DAL/Repositories/Implementations/UserRepository.cs [hunk 2]
```diff
@@ -1,17 +1,47 @@
...
+using OnlineShop.DAL.Infrastructure;
 using OnlineShop.DAL.Repositories.Interfaces;
 
 namespace OnlineShop.DAL.Repositories.Implementations;
 
-public class UserRepository:IBaseRepository<User>
+public class UserRepository(ShopDbContext dbContext) : IUserRepository
 {
-    public Task<User> GetById(Guid id)
+    public async Task<User> GetById(Guid id)
```
Comment (mebry) [inline OnlineShop/OnlineShop.DAL/Repositories/Implementations/UserRepository.cs]: For asynchronous methods, pass CancellationToken and use it

Comment (mebry) [inline OnlineShop/OnlineShop.DAL/Repositories/Implementations/UserRepository.cs]: Always add the suffix "Async" to the names of methods that have an Async or async modifie

File: OnlineShop/OnlineShop.DAL/Repositories/Implementations/UserRepository.cs [hunk 3]
```diff
@@ -1,17 +1,47 @@
...
     {
-        throw new NotImplementedException();
+        var user = await dbContext.Users.SingleOrDefaultAsync(u => u.Id.Equals(id));
```
Comment (mebry) [inline OnlineShop/OnlineShop.DAL/Repositories/Implementations/UserRepository.cs]: Can be simplified

**Review by mebry — CHANGES_REQUESTED** [review]

**Review by mebry — APPROVED** [review]

**Pull request merged**